content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md

```diff
@@ -0,0 +1,628 @@
+
+
+## t-SNE + Distance Metrics = Visualization of Similarity Between Texts
+
+To tackle the problem of detecting variations in drug names, we can use text distance metrics like Levenshtein distance, which measures the edits needed to transform one string into another. By calculating these distances, we can identify subtle differences. However, a distance matrix can be difficult to interpret. This is where t-SNE (t-distributed Stochastic Neighbor Embedding) is useful. It converts high-dimensional distance matrices into a 2D visualization, clustering similar items together. For example, typos like “Ibuprofin” and “Ibuprofine” will appear close together, making it easier to spot inconsistencies in drug names.
+
+This approach transforms the complex task of detecting textual variations into a visual pattern-recognition challenge. Drug names with multiple typos or spellings cluster together; for instance, “Omeprazole” and its variations will form distinct groups, while names like “Warfarin” and “Warferin” will cluster closely. This visualization reveals patterns that are hard to discern in tables, such as recurring typos for specific drugs or misspellings used by different researchers.
+
+Moreover, t-SNE can highlight complex patterns, like regional spelling differences (e.g., “Amoxicillin” vs. “Amoxycillin”). While traditional text-matching tools may miss these nuances, t-SNE effectively shows which names are similar based on their edit distances. When combined with interactive visualization tools like Plotly or ggplot, users can hover over data points to see original names, facilitating the exploration of clusters and enabling direct corrections. This method not only streamlines data cleaning but is crucial for maintaining drug name accuracy in clinical research, where precision is vital.
+
+
+## Clustering - That Is, Group Similar Into Similar
+
+The graphics can be tricky to interpret, especially when we have a lot of drugs. That's why we use an algorithmic approach based on clustering "similar into similar" in our workflow. One clustering technique we find really effective is DBSCAN (Density-Based Spatial Clustering of Applications with Noise). This algorithm is great at identifying clusters of different shapes and sizes, which helps us find meaningful patterns in the complex landscape of drug names. By looking at how dense the data is, DBSCAN groups similar drug names together while also showing us any outliers that might be typos. For example, it can group entries like "Ibuprofin" and "Ibuprofine," showing us any potential inconsistencies that could affect clinical trial results. This powerful approach not only makes the data more reliable but also makes it easier to clean up, so researchers can focus on getting good results in their clinical studies.
+
+
+# Case Study: Validation of Drug Names for a Pharmaceutical Gig
```

Comment:
I was wondering if we might want to consider either simplifying headers or possibly removing them altogether. Since we’re aiming for a ‘serious friend’ tone, it might work better to let the content speak for itself without too much emphasis on sub-headers. What do you think?